src/checkpoint/container/view_serialize.h

```diff
@@ -313,9 +313,17 @@ inline void serialize_impl(SerializerT& s, Kokkos::DynRankView<T,Args...>& view)
     serializeLayout<SerializerT>(s, 7, layout_cur);
   }
 
+  // Serialize the total number of elements in the Kokkos::View
+  size_t num_elms = view.size();
+  s | num_elms;
+  bool is_uninitialized = dims == 0 and num_elms == 0;
   // Construct a view with the layout and use operator= to propagate out
   if (s.isUnpacking()) {
-    if (dims == 1) {
+    if (is_uninitialized) {
+      view = ViewType{};
+    } else if (dims == 0) {
+      view = constructRankedView<ViewType,0>(label, std::make_tuple(layout));
+    } else if (dims == 1) {
```

Comment:
We may be able to simplify a bunch of the view reconstruction logic once Kokkos 3.7 ships, since it now properly round-trips the layout associated with a View(https://github.com/kokkos/kokkos/pull/5188), and DRV will do the same https://github.com/kokkos/kokkos/pull/5026

Reply:
https://github.com/DARMA-tasking/magistrate/issues/254 to track  this